demo/kserve/scripts/utils.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+
+die() {
+  color_red='\e[31m'
+  color_yellow='\e[33m'
+  color_reset='\e[0m'
+  printf "${color_red}FATAL:${color_yellow} $*${color_reset}\n" 1>&2
+  exit 10
+}
+
+info() {
+  color_blue='\e[34m'
+  color_reset='\e[0m'
+  printf "${color_blue}$*${color_reset}\n" 1>&2
+}
+
+success() {
+  color_green='\e[32m'
+  color_reset='\e[0m'
+  printf "${color_green}$*${color_reset}\n" 1>&2
+}
+
+check_pod_status() {
+  local -r JSONPATH="{range .items[*]}{'\n'}{@.metadata.name}:{@.status.phase}:{range @.status.conditions[*]}{@.type}={@.status};{end}{end}"
+  local -r pod_selector="$1"
+  local -r pod_namespace="$2"
+  local pod_status
+  local pod_entry
+
+  pod_status=$(oc get pods -l $pod_selector -n $pod_namespace -o jsonpath="$JSONPATH") 
+  oc_exit_code=$? # capture the exit code instead of failing
+
+  if [[ $oc_exit_code -ne 0 ]]; then
+    # kubectl command failed. print the error then wait and retry
+    echo "Error running kubectl command."
+    echo $pod_status
+    return 1
+  elif [[ ${#pod_status} -eq 0 ]]; then
+    echo -n "No pods found with selector $pod_selector in $pod_namespace. Pods may not be up yet."
+    return 1
+  else
+    # split string by newline into array
+    IFS=$'\n' read -r -d '' -a pod_status_array <<<"$pod_status"
+
+    for pod_entry in "${pod_status_array[@]}"; do
+      local pod=$(echo $pod_entry | cut -d ':' -f1)
+      local phase=$(echo $pod_entry | cut -d ':' -f2)
+      local conditions=$(echo $pod_entry | cut -d ':' -f3)
+      if [ "$phase" != "Running" ] && [ "$phase" != "Succeeded" ]; then
+        return 1
+      fi
+      if [[ $conditions != *"Ready=True"* ]]; then
+        return 1
+      fi
+    done
+  fi
+  return 0
+}
+
+wait_for_pods_ready() {
+  local -r JSONPATH="{.items[*]}"
+  local -r pod_selector="$1"
+  local -r pod_namespace=$2
+  local wait_counter=0
+  local oc_exit_code=0
+  local pod_status
+
+  while true; do
+    pod_status=$(oc get pods -l $pod_selector -n $pod_namespace -o jsonpath="$JSONPATH") 
+    oc_exit_code=$? # capture the exit code instead of failing
+
+    if [[ $oc_exit_code -ne 0 ]]; then
+      # kubectl command failed. print the error then wait and retry
+      echo $pod_status
+      echo -n "Error running kubectl command."
+    elif [[ ${#pod_status} -eq 0 ]]; then
+      echo -n "No pods found with selector '$pod_selector' -n '$pod_namespace'. Pods may not be up yet."
+    elif check_pod_status "$pod_selector" "$pod_namespace"; then
+      echo "All $pod_selector pods in '$pod_namespace' namespace are running and ready."
+      return
+    else
+      echo -n "Pods found with selector '$pod_selector' in '$pod_namespace' namespace are not ready yet."
+    fi
+
+    if [[ $wait_counter -ge 60 ]]; then
+      echo
+      oc get pods $pod_selector -n $pod_namespace
```

Comment:
```
oc get pods -l $pod_selector -n $pod_namespace
```
otherwise:
```
+ oc get pods serving.kserve.io/inferenceservice=caikit-example-isvc -n kserve-demo
error: there is no need to specify a resource type as a separate argument when passing arguments in resource/name form (e.g. 'oc get resource/<resource_name>' instead of 'oc get resource resource/<resource_name>'
```
CI-tested ™️ :D

Reply:
updated